File: app/src/main/java/com/project/scratchstudio/kith_andoid/Adapters/TreeAdapter.java
package com.project.scratchstudio.kith_andoid.Adapters;

import android.app.Activity;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.project.scratchstudio.kith_andoid.Holders.TreeHolder;
import com.project.scratchstudio.kith_andoid.network.model.User;
import com.project.scratchstudio.kith_andoid.R;
import com.project.scratchstudio.kith_andoid.Service.PicassoCircleTransformation;
import com.squareup.picasso.MemoryPolicy;
import com.squareup.picasso.Picasso;

import java.util.List;

public class TreeAdapter extends RecyclerView.Adapter<TreeHolder> {

    public interface OnItemClickListener {
        void onItemClick(User item);
    }

    private List<User> userList;
    private final TreeAdapter.OnItemClickListener listener;
    private Activity activity;

    public TreeAdapter(Activity activity, List<User> annInfos, TreeAdapter.OnItemClickListener listener) {
        this.activity = activity;
        this.userList = annInfos;
        this.listener = listener;
    }

    @NonNull
    @Override
    public TreeHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.list_item_layout, viewGroup, false);
        return new TreeHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull TreeHolder holder, int i) {
        User user = userList.get(i);
        String name = user.getFirstName() + " " + user.getLastName();
        holder.name.setText(name);
        holder.position.setText(user.getPosition());

            Picasso.with(activity).load(user.getUrl().replaceAll("@[0-9]*", ""))
                    .placeholder(com.project.scratchstudio.kith_andoid.R.mipmap.person)
                    .error(com.project.scratchstudio.kith_andoid.R.mipmap.person)
                    .transform(new PicassoCircleTransformation())
                    .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
                    .into(holder.image);

        holder.bind(user, listener);
    }

    @Override
    public int getItemCount() {
        return userList.size();
    }

}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/adapters/TreeAdapter.kt
package com.project.scratchstudio.kith_andoid.adapters

import android.app.Activity
import android.view.LayoutInflater
import android.view.ViewGroup

import com.project.scratchstudio.kith_andoid.holders.TreeHolder
import com.project.scratchstudio.kith_andoid.R
import com.project.scratchstudio.kith_andoid.service.PicassoCircleTransformation
import com.project.scratchstudio.kith_andoid.network.model.user.User
import com.squareup.picasso.MemoryPolicy
import com.squareup.picasso.Picasso
import androidx.recyclerview.widget.RecyclerView

class TreeAdapter(private val activity: Activity, private var userList: List<User>?, private val listener: OnItemClickListener) : RecyclerView.Adapter<TreeHolder>() {

    interface OnItemClickListener {
        fun onItemClick(item: User)
    }

    fun setUserList(list: List<User>) {
        userList = list
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): TreeHolder {
        val itemView = LayoutInflater.from(viewGroup.context).inflate(R.layout.list_item_layout, viewGroup, false)
        return TreeHolder(itemView)
    }

    override fun onBindViewHolder(holder: TreeHolder, i: Int) {
        val user = userList!![i]
        val name = user.firstName + " " + user.lastName
        holder.name.text = name
        holder.position.text = user.position

        Picasso.with(activity).load(user.photo!!.replace("@[0-9]*".toRegex(), ""))
                .placeholder(R.mipmap.person)
                .error(R.mipmap.person)
                .transform(PicassoCircleTransformation())
                .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
                .into(holder.image)

        holder.bind(user, listener)
    }

    override fun getItemCount(): Int {
        return userList!!.size
    }

}
